File: 백준/Silver/11399. ATM/ATM.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader input = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(input.readLine());
        String[] inputArr = input.readLine().split(" ");
        int[] arr = new int[n];
        for (int i = 0; i < n; i++) {
            int time = Integer.parseInt(inputArr[i]);
            arr[i] = time;
        }
        Arrays.sort(arr);
        int tmp = 0;
        int result = 0;
        for (int i = 0; i < n; i++) {
            tmp = tmp + arr[i];
            result += tmp;
        }
        System.out.println(result);

    }
}

File: 백준/Silver/11399. ATM/ATM.py
import sys

input = sys.stdin.readline

N = int(input())

P = list(map(int, input().split()))
dic = {}
for i in range(1,N+1):
    dic[i] = P[i-1]

dic = dict(sorted(dic.items(), key=lambda x:x[1]))
previous_time = 0
time = 0
for i in dic.values():
    previous_time += i
    time += previous_time
print(time)

